dev/sg/sg_src.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+	"context"
+	"os"
+
+	"github.com/sourcegraph/sourcegraph/dev/sg/internal/secrets"
+	"github.com/sourcegraph/sourcegraph/dev/sg/internal/std"
+	"github.com/sourcegraph/sourcegraph/dev/sg/internal/usershell"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"github.com/sourcegraph/sourcegraph/lib/output"
+	"github.com/urfave/cli/v2"
+)
+
+type srcInstance struct {
+	AccessToken string `json:"access_token"`
+	Endpoint    string `json:"endpoint"`
+}
+
+type srcSecrets struct {
+	Instances map[string]srcInstance `json:"instances"`
+}
+
+var srcCommand = &cli.Command{
+	Name:      "src",
+	UsageText: "sg src [instance] [src-cli args]",
+	Usage:     "Run src-cli on a given instance",
+	Category:  CategoryCompany,
+	Action: func(cmd *cli.Context) error {
+		_, sc, err := getSrcSecret(cmd.Context, std.Out)
+		if err != nil {
+			return err
+		}
+		instance, ok := sc.Instances[cmd.Args().Get(0)]
+		if !ok {
+			std.Out.WriteFailuref("Instance not found, register one with 'sg src register-instance'")
+			return errors.New("instance not found")
+		}
+		srcArgs := cmd.Args().Slice()[1:]
+		c := usershell.Command(cmd.Context, append([]string{"src"}, srcArgs...)...)
+		c = c.Env(map[string]string{
+			"SRC_ACCESS_TOKEN": instance.AccessToken,
+			"SRC_ENDPOINT":     instance.Endpoint,
+		})
+		return c.Run().Stream(os.Stdout)
+	},
+	Subcommands: []*cli.Command{
+		{
+			Name:      "instance",
+			UsageText: "todo",
+			Subcommands: []*cli.Command{
+				{
+					Name:    "register",
+					Usage:   "",
+					Aliases: []string{"update, add"},
+					Flags: []cli.Flag{
+						&cli.StringFlag{
+							Name:     "name",
+							Usage:    "Name for the new instance (ex: s2)",
+							Required: true,
+						},
+						&cli.StringFlag{
+							Name:     "endpoint",
+							Usage:    "Endpoint for the new instance (ex: https://sourcegraph.sourcegraph.com)",
+							Required: true,
+						},
+						&cli.StringFlag{
+							Name:     "access-token",
+							Usage:    "AccessToken for the new instance",
```

Comment:
To allow setting from env:

```suggestion
							Name:     "access-token",
							Usage:    "access token for the new instance",
							EnvVars: []string{"SRC_ACCESS_TOKEN"},
```

Reply:
The access token alone isn't sufficient to connect to another instance (you need the endpoint to be correct too) - perhaps `SG_SRC_ACCESS_TOKEN` instead? Additionally I think the flag takes precedence over the env var regardless, so if you set the flag you won't have a conflict

Reply:
Ah, if flags takes precedence, that might work. I'll check that 🙏

Reply:
Agree with JH here. I don't want my ENV vars to take precedence here or be used at all. That's the whole point of using `sg src`, otherwise I could just call `src` directly.